src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -223,141 +192,52 @@ private void testPluginAgainst(
-            throw new UncheckedIOException(e);
-        }
-
-        // Ran the BeforeCompileHooks
+        // Run the BeforeCompileHooks
         BeforeCompilationContext beforeCompile =
-                new BeforeCompilationContext(plugin, model, coreVersion, config, pluginCheckoutDir, parentFolder);
+                new BeforeCompilationContext(pluginMetadata, coreVersion, config, cloneLocation);
         pcth.runBeforeCompilation(beforeCompile);
 
         // First build against the original POM. This defends against source incompatibilities
         // (which we do not care about for this purpose); and ensures that we are testing a
         // plugin binary as close as possible to what was actually released. We also skip
         // potential javadoc execution to avoid general test failure.
-        if (!beforeCompile.ranCompile()) {
-            runner.run(
-                    Map.of("maven.javadoc.skip", "true"),
-                    pluginCheckoutDir,
-                    buildLogFile,
-                    "clean",
-                    "process-test-classes");
-        }
+        // as we may need to resolve artifacts of incrementals or CD releases ensure we
+        // use the same version so parents and other dependencies can be found
+        // if not an incremental this property is ignored so is safe to set.
+        runner.run(
+                Map.of(
+                        "maven.javadoc.skip", "true",
+                        "set.changelist", "true"),
+                cloneLocation,
+                pluginMetadata.getModulePath(),
+                buildLogFile,
+                "clean",
+                "process-test-classes");
 
         List<String> args = new ArrayList<>();
         args.add("hpi:resolve-test-dependencies");
         args.add("hpi:test-hpl");
         args.add("surefire:test");
 
         // Run preexecution hooks
-        BeforeExecutionContext forExecutionHooks = new BeforeExecutionContext(
-                plugin,
-                model,
-                coreVersion,
-                config,
-                pluginCheckoutDir,
-                parentFolder,
-                args,
-                new MavenPom(pluginCheckoutDir));
+        BeforeExecutionContext forExecutionHooks =
+                new BeforeExecutionContext(pluginMetadata, coreVersion, config, cloneLocation, args);
         pcth.runBeforeExecution(forExecutionHooks);
 
         Map<String, String> properties = new LinkedHashMap<>(config.getMavenProperties());
         properties.put("overrideWar", config.getWar().toString());
         properties.put("jenkins.version", coreVersion);
         properties.put("useUpperBounds", "true");
+        // enable setting the incremental / CD release so if a multi-module reactor
+        // any other projects can be obtained from the repo
+        properties.put("set.changelist", "true");
+        // as hooks may be adjusting the POMs tell the git-changelist extension to ignore dirty
```

Comment:
:eyes:

Reply:
it is legal (there are no requirements to say otherwise) and whilst the OSS ones have been re-written to use properties - some proprietary ones exist that manipulate the pom (e.g. CloudBees' wars have an extra jar in `WEB-INF/lib` of the war that we need to update for plugins that depend on it and as it is not in OSS Jenkins nor is a plugin the `maven-hpi-plugin` is unaware of it.